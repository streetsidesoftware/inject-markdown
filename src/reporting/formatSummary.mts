import chalkDefault from 'chalk';

import type { Result } from '../processor/process.mjs';

export function formatSummary(r: Result, chalk = chalkDefault): string {
    return `\
Summary:
    Number of files:                    ${r.numberOfFiles}
    Number of files processed:          ${r.numberOfFilesProcessed}
    Number of files with injections:    ${r.numberOfFilesWithInjections}
    Number of files written:            ${r.numberOfFilesWritten}
    Number of files skipped:            ${
        r.numberOfFilesSkipped ? chalk.yellow(r.numberOfFilesSkipped) : r.numberOfFilesSkipped
    }
    Errors:                             ${r.errorCount ? chalk.red(r.errorCount) : r.errorCount}
`;
}
