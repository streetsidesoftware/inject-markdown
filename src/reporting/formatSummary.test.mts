import { Chalk } from 'chalk';
import { describe, expect, test, beforeEach } from 'vitest';
import { formatSummary } from './formatSummary.mjs';

const chalk = new Chalk();

describe('formatSummary', () => {
    beforeEach(() => {
        chalk.level = 0;
    });
    test('formatSummary', () => {
        const result = {
            numberOfFiles: 0,
            numberOfFilesProcessed: 0,
            numberOfFilesWithInjections: 0,
            numberOfFilesUpdated: 0,
            numberOfFilesWritten: 0,
            numberOfFilesSkipped: 0,
            filesWithErrors: [],
            errorCount: 0,
        };
        expect(formatSummary(result, chalk)).toMatchSnapshot();
    });

    test('formatSummary with errors', () => {
        const result = {
            numberOfFiles: 1,
            numberOfFilesProcessed: 1,
            numberOfFilesWithInjections: 1,
            numberOfFilesUpdated: 0,
            numberOfFilesWritten: 0,
            numberOfFilesSkipped: 1,
            filesWithErrors: ['README.md'],
            errorCount: 1,
        };
        expect(formatSummary(result, chalk)).toMatchSnapshot();
    });
});
