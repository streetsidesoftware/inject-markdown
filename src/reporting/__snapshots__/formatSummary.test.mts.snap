// Vitest Snapshot v1

exports[`formatSummary > formatSummary 1`] = `
"Summary:
    Number of files:                    0
    Number of files processed:          0
    Number of files with injections:    0
    Number of files written:            0
    Number of files skipped:            0
    Errors:                             0
"
`;

exports[`formatSummary > formatSummary with errors 1`] = `
"Summary:
    Number of files:                    1
    Number of files processed:          1
    Number of files with injections:    1
    Number of files written:            0
    Number of files skipped:            1
    Errors:                             1
"
`;
